Source/WarpX.cpp

```diff
@@ -853,6 +853,13 @@ WarpX::ReadParameters ()
         pp_warpx.query("do_pml_j_damping", do_pml_j_damping);
         pp_warpx.query("do_pml_in_domain", do_pml_in_domain);
         pp_warpx.query("do_similar_dm_pml", do_similar_dm_pml);
+        // Read `v_particle_pml` in units of the speed of light
+        v_particle_pml = 1;
+        pp_warpx.query("v_particle_pml", v_particle_pml);
+        // Scale by the speed of light
+        v_particle_pml = v_particle_pml * PhysConst::c;
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(v_particle_pml > 0._rt,
+            "input value for the velocity warpx.v_particle_pml of the macroparticle must be positive.");
```

Comment:
Maybe here it would make sense to check that the user's input `v_particle_pml` is such that both `0._rt < v_particle_pml` and `v_particle_pml <= 1._rt`, since a case `v_particle_pml > c` (after scaling by the speed of light) would not be acceptable.

I think something like this could work, to be done _before_ scaling `v_particle_pml` by the speed of light `c` (note also that we would need to update the error message, as in the following suggestion):
```cpp
        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(0._rt < v_particle_pml && v_particle_pml <= 1._rt,
            "Input value for the velocity warpx.v_particle_pml of the macroparticle must be in (0,1] (in units of c).");
```